certbot/client.py

```diff
@@ -362,6 +362,48 @@ def save_certificate(self, certr, chain_cert,
 
             return abs_cert_path, abs_chain_path, abs_fullchain_path
 
+    def renew_deploy_certificate(self, domains, privkey_path,
+                           cert_path, chain_path, fullchain_path):
+        """Allow installer to handle renewed certificate
+
+        :param list domains: list of domains to install the certificate
+        :param str privkey_path: path to certificate private key
+        :param str cert_path: certificate file path (optional)
+        :param str chain_path: chain file path
+
+        """
+        if self.installer is None:
+            logger.warning("No installer specified, client is unable to deploy"
+                           "the certificate")
+            raise errors.Error("No installer available")
+
+        chain_path = None if chain_path is None else os.path.abspath(chain_path)
+
+        with error_handler.ErrorHandler(self.installer.recovery_routine):
+            for dom in domains:
+                # this is an optional method. be sure it exists
+                # before executing
+                if (hasattr(self.installer, 'renew_deploy_cert') and
```

Comment:
https://jeffknupp.com/blog/2013/02/06/write-cleaner-python-use-exceptions/

Or, in short, it's more Pythonic to write this as `try: except:` than to test if it has the method before calling.